### ISSUE TITLE
[BUG] Zone update dont work if the record is in a clone zone where the Domain Name is the same

### ISSUE THREAD

**Issue by JsdBEnZ7Rt9orgOfa1Ep1e4n** [WillyXJ/facileManager#596]
Replace everything between stars with current version of your facileManager and module installations:  
fM Version : **v4.5.0**
fmDNS  Version : **v5.3.3 **

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

**(BUG | ISSUE) Expected Behavior:**
Update record in the client with the php cron command
```
/usr/bin/php /usr/local/facileManager/fmDNS/client.php cron 
```

**(BUG | ISSUE) Actual Behavior:**
Error on the server
```
PHP Fatal error:  Uncaught Error: Cannot use object of type stdClass as array in /opt/facileManager/server/fm-modules/fmDNS/classes/class_buildconf.php:1094\nStack trace:\n#0 /opt/facileManager/server/fm-modules/shared/pages/buildconf.php(57): fm_module_buildconf->buildCronConfigs()\n#1 /opt/facileManager/server/index.php(57): include('...')\n#2 {main}\n  thrown in /opt/facileManager/server/fm-modules/fmDNS/classes/class_buildconf.php on line 1094
```

**(BUG | ISSUE) Steps to reproduce:**
Create a forward zone  "example.lan" on the view "view1" 
Create a clone of the zone "example.lan" with same domain name on the view "view2"

Result : 

![Screenshot 2023-12-11 at 18-14-22 Forward Zones ‹ facileManager](https://github.com/WillyXJ/facileManager/assets/146933594/c947099d-f430-450f-b5c4-0f8deb27b145)

Add a record in the first or the second zone.
Sync on the client with the cron et get the error on the server

```
PHP Fatal error:  Uncaught Error: Cannot use object of type stdClass as array in /opt/facileManager/server/fm-modules/fmDNS/classes/class_buildconf.php:1094\nStack trace:\n#0 /opt/facileManager/server/fm-modules/shared/pages/buildconf.php(57): fm_module_buildconf->buildCronConfigs()\n#1 /opt/facileManager/server/index.php(57): include('...')\n#2 {main}\n  thrown in /opt/facileManager/server/fm-modules/fmDNS/classes/class_buildconf.php on line 1094
```

In addition, there is no error, if i use the "BUILD CONFIG" in the https://xxxx/dnsManage/config-servers.php page.

The error is only when I use the **Process update** available in the main page.

We upgraded from facileManager v3.5.4 and made some test with php 7.3 and php 8.2 (same issue)

Thanks for support

**Comment by JsdBEnZ7Rt9orgOfa1Ep1e4n** [comment]
This commit seem resolve the issue
https://github.com/JsdBEnZ7Rt9orgOfa1Ep1e4n/facileManager/commit/ebddc5c21b2df4468742c16eebd8d6e40cf370af

**Comment by generobruno** [comment]
After modifying the code to match that commit i have another error. If i clone the zone and do not add any more records i get:

 PHP Fatal error:  Uncaught Error: Attempt to assign property "record_name" on null in /var/www/html/dnsadmin/fm-modules/fmDNS/classes/class_buildconf.php:1247

However, it works fine when i add a test record to the cloned zone.

**Comment by WillyXJ** [comment]
This is now fixed in [fmDNS 6.0.0](http://www.facilemanager.com/modules/fmdns/) and later.

**Comment by JsdBEnZ7Rt9orgOfa1Ep1e4n** [comment]
Thanks for the fix
